### PR TITLE
Add a slide on overloading

### PR DESCRIPTION
**Pull request by bernhardmgruber**

### CODE REVIEW

File: talk/basicconcepts/functions.tex [hunk 1]
```diff
@@ -277,3 +277,28 @@
...
+            \item A different return type alone is not allowed
+            \item Default arguments can cause ambiguities
+            \item All functions with the same name form an ``overload set''
+        \end{itemize}
+    \end{block}
+    \begin{exampleblock}{}
+        \begin{cppcode*}{gobble=6}
+            int sum(int b);             // 1
+            int sum(int b, int c);      // 2, ok, overload
+            // float sum(int b, int c); // would error
```
Comment (hageboeck) [inline talk/basicconcepts/functions.tex:296]:
```suggestion
            // float sum(int b, int c); // disallowed
```

**Review by hageboeck — APPROVED** [review]
LGTM, but maybe replace the one comment

**Review by sponce — APPROVED** [review]
I would also move that slide to before the exercise and the good practice ones.

**Comment by bernhardmgruber** [comment]
> I would also move that slide to before the exercise and the good practice ones.

done.

**Pull request merged**